File: src/sweepy-noise.hs
-- sweepy noise (jmcc)

import Sound.SC3.ID

sweepy_noise :: UGen
sweepy_noise =
    let n = uclone 'a' 2 (whiteNoise 'a' AR)
        lfoDepth = mouseY' KR 200 8000 Exponential 0.1
        lfoRate = mouseX' KR 4 60 Exponential 0.1
        freq = lfSaw KR lfoRate 0 * lfoDepth + (lfoDepth * 1.2)
        filtered = rlpf (n * 0.03) freq 0.1
    in combN filtered 0.3 0.3 2 + filtered

main :: IO ()
main = audition (out 0 sweepy_noise)

File: gr/jmcc/sweepy-noise.fs
\ sweepy noise (jmcc) #6

: x 4 60 1 0.1 MouseX.kr ;
: y 200 8000 1 0.1 MouseY.kr ;
: f WhiteNoise.ar 2 clone 0.03 * x 0 LFSaw.kr y * y 1.2 * + 0.1 RLPF ;
f 0.3 0.3 2 CombN f + play
